src/Util.vala

```diff
@@ -166,23 +166,60 @@ namespace Tasks.Util {
     //--- X-Property ---//
 
 
-    public unowned ICal.Property? get_icalcomponent_x_property (ICal.Component ical_component, string x_property_name) {
+    public ICal.Property? get_icalcomponent_x_property (ICal.Component ical_component, string x_property_name) {
         return get_ecalpropertybag_x_property (new ECal.ComponentPropertyBag.from_component (ical_component, (property) => {
             return property.isa () == ICal.PropertyKind.X_PROPERTY;
         }), x_property_name);
     }
 
-    public unowned ICal.Property? get_ecalpropertybag_x_property (ECal.ComponentPropertyBag ecal_propertybag, string x_property_name) {
-        unowned ICal.Property? property = null;
-
+    public ICal.Property? get_ecalpropertybag_x_property (ECal.ComponentPropertyBag ecal_propertybag, string x_property_name) {
         var ecal_propertybag_count = ecal_propertybag.get_count ();
+
         for (int i = 0; i < ecal_propertybag_count; i++) {
-            property = ecal_propertybag.get (i);
+            unowned ICal.Property? property = ecal_propertybag.get (i);
             if (property.isa () == ICal.PropertyKind.X_PROPERTY && property.get_x_name () == x_property_name) {
-                break;
+                var x_property = new ICal.Property (ICal.PropertyKind.X_PROPERTY);
+                x_property.set_x_name (x_property_name);
+
+                if (property.get_x () != null) {
+                    x_property.set_x (property.get_x ().dup ());
+                }
+
+                return x_property;
+            }
+        }
+        return null;
+    }
+
+
+    //--- Sort ---//
+
+
+    public string get_ecalcomponent_sort (ECal.Component ecalcomponent) {
+        var sort = "";
+
+        unowned ICal.Component? icalcomponent = ecalcomponent.get_icalcomponent ();
+        if (icalcomponent != null) {
+            var x_apple_sort_order_property = get_icalcomponent_x_property (icalcomponent, "X-APPLE-SORT-ORDER");
+
+            if (x_apple_sort_order_property != null ) {
+                sort = x_apple_sort_order_property.get_x ();
             }
         }
-        return property;
+
+        if (sort == "") {
+            /**
+            * if the property is not set, Apple uses the
+            * number of seconds between the creation of
+            * the task and 20010101T000000Z (Cocoa/Webkit epoch)
+            * as value for sorting.
+            */
+            sort = ecalcomponent.get_created ()
+            .subtract (new ICal.Time.from_string ("20010101T000000Z"))
+            .as_int ()
+            .to_string ();
+        }
```

Comment:
Hrm, it seems weird to convert to string here and it seems the only reason that's being done is because the value for `X-APPLE-SORT-ORDER` is a string. It might be better to have this function only do one thing (get the apple sort order) and handle the null case in the listbox sort function still

Reply:
Created a new `public ICal.Duration get_apple_sortorder_default_value` in `Tasks.Util`, which returns the default value. Converting to the desired format is then done in the listbox sort function.